lib/rspec/mocks/space.rb

```diff
@@ -143,10 +144,11 @@ def any_instance_recorders_from_ancestry_of(object)
     private
 
       # We don't want to depend on the stdlib ourselves, but if the user is
-      # using threads then a Mutex will be available to us. If not, we don't
-      # need to synchronize anyway.
+      # using threads then a Mutex will be available to us (which we use via
+      # (RSpec::Support::ReentrantMutex). If not, we don't need to synchronize
+      # anyway.
       def new_mutex
-        defined?(::Mutex) ? ::Mutex.new : FakeMutex
+        defined?(::Mutex) ? Support::ReentrantMutex.new : FakeMutex
```

Comment:
I don't think the `define?(::Mutex)` bit or `FakeMutex` makes sense here anymore.